### DISCUSSION TITLE
Precompiled version with filament sensor support

### DISCUSSION THREAD

**Discussion by melco007** [Jyers/Marlin#131] [Ideas]
This firmware seems  very very good to me. I am currently controlling my endrer 3 v2 with the firmware from Smith3D and the only reason that prevents me from upgrading to this one is that the precompiled version does not seem to support the filament detector yet (Bigtreetech) in my case.  Will it be possible to have a precompiled version with filament sensor?

**Comment by swbog** [comment]
See discussion about not asking for custom precompiled configs for every scenario. You could always configure yourself and add the feature

**Reply by melco007** [reply]
OK sorry. I will go for it and learn. you are right. However, allow me a quick question. in the configuration folder. on what basis should I go for my personal feature, Jyers's Config or E3V2 Templates?

**Reply by Schnabulation** [reply]
Thank you for understanding. It is really easy to compile your own firmware and you should be able to learn it in less than an hour - and it will allow you to customize your printer even more.

As for your question: I was using my own Ender 3 V2 configuration files with Jyers firmware and that worked flawlessly. If you start from scratch, just use one of Jyers configuration.

**Reply by melco007** [reply]
Great, thank you!

**Reply by Schnabulation** [reply]
If you need any help regarding compiling your own firmware, just hit me up directly - I can help you, ok?

**Reply by melco007** [reply]
Thanks, I compiled my own version this morning and it works !! I still have a few tests to do, but everything looks ok at the moment.

**Comment by Wanderingnlost** [comment]
So Im trying to do the same, activate filament senor. Trying to follow youtube but Im confused on one thing they do not show.  I believe I downloaded the code correctly but there are multiple config files. i was going to utilize e32 template manualmesh 3v3.  Did you delete the other files. ex bltouch config?  First attempt at doing this.  Seems like multiple config files would mess things up

**Reply by Henri-J-Norden** [reply]
You have to replace the configuration.h file in the [Marlin ](https://github.com/Jyers/Marlin/tree/Ender_3_V2_Extensible_UI/Marlin) folder with the one you want to use.
